### ISSUE TITLE
Why don't you implements onAttached, onDetached events for virtual dom element

### ISSUE THREAD

**Issue by cevek** [joelrich/citojs#9]
```javascript
cito.vdom.append({tag: "div", events: {attached: function(){ console.log("attached") }}})
```

**Comment by joelrich** [comment]
There is already a $created, $destroyed and $changed event, but it's not yet documented and will likely change again.

**Comment by xorcus** [comment]
Would "webcomponents.org" and custom-element callbacks do as a concrete feedback? It's similar to $created, $destroyed and $changed you mentioned, just named differently:
    http://www.w3.org/TR/custom-elements/#types-of-callbacks
I am just wondering whether attributeChangedCallback gets invoked only when an element attribute changes, or it also gets invoked when an element (JavaScript object) property changes

**Comment by joelrich** [comment]
attributeChangedCallback is only called for attributes. I could rename them to $attached, $detached and $changed. The $-prefix is to distinguish them from real events.

**Comment by xorcus** [comment]
I believe you have $created and $attached as two separate events, because API has create() and append() methods. The first one would fire only $created, while the second one would fire both $created and $attached. One may want to know when the component is in-dom to e.g. set the focus, or examine the rendered size of an element (which is usually 0 before the element get rendered)

**Comment by joelrich** [comment]
Ah right, now I remember one of the problems with these events: Due to some browser-specific performance optimisations, $created is triggered before the element is attached to the DOM in Chrome and Firefox, but after in IE. Will have to check if it makes a difference to use the same order in all browsers.

**Comment by mneumann** [comment]
@joelrich ```$created``` is called before the node is attach to the DOM. The problem with that is, that elements doesn't have a size yet, so for example, when you want to use it to change the layout (think of masonry style), it is not possible. What I am doing as a workaround atm, is to do the layout changes after ```cito.vdom.update()``` returns, which is perfectly fine for me :)

**Comment by joelrich** [comment]
@mneumann I am aware of the problem and will fix it, but first I am trying to get the rewrite done (which takes much longer than I hoped).

**Comment by mildred** [comment]
I noticed that the `$destroyed` event is not recursive. making it recursive would allow some flexibility. See for example this test: https://gist.github.com/mildred/26e319928b98b32d5f851d80c580e086

**Comment by joelrich** [comment]
@mildred that's right. On the other hand, it would have been dangerous for the performance (see https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Mutation_events as an analogy). I tried to make these events very lightweight which is also the reason why they do not bubble. They should probably not have been part of the `events` but a separate `hooks` object.
